source/E2E.Tests/Services/BesiegerCamps/BesiegerCampCollectionTests.cs

```diff
@@ -0,0 +1,172 @@
+                BesiegerCampCollectionPatches.ListAddOverride(besiegerCamp._besiegerParties, besiegerParty, besiegerCamp);
+            });
+
+            //Assert
+            foreach (var client in Clients)
+            {
+                Assert.True(client.ObjectManager.TryGetObject<BesiegerCamp>(besiegerCampId, out var clientBesiegerCamp));
+                Assert.True(client.ObjectManager.TryGetObject<MobileParty>(besiegerPartyId, out var clientBesiegerParty));
+                Assert.Contains<MobileParty>(clientBesiegerParty, clientBesiegerCamp._besiegerParties);
+            }
+        }
+
+        [Fact]
+        private void ServerRemoveBesiegerParty_SyncAllClients()
+        {
+            // Arrange
+            var besiegerCamp = ServerCreateObject<BesiegerCamp>(out string besiegerCampId);
+            var besiegerParty = ServerCreateObject<MobileParty>(out string besiegerPartyId);
+            Server.Call(() =>
+            {
+                BesiegerCampCollectionPatches.ListAddOverride(besiegerCamp._besiegerParties, besiegerParty, besiegerCamp);
+            });
+            foreach (var client in Clients)
+            {
+                Assert.True(client.ObjectManager.TryGetObject<BesiegerCamp>(besiegerCampId, out var clientBesiegerCamp));
+                Assert.True(client.ObjectManager.TryGetObject<MobileParty>(besiegerPartyId, out var clientBesiegerParty));
+                Assert.Contains(clientBesiegerParty, clientBesiegerCamp._besiegerParties);
+            }
+
+            // Act
+            Server.Call(() =>
+            {
+                BesiegerCampCollectionPatches.ListRemoveOverride(besiegerCamp._besiegerParties, besiegerParty, besiegerCamp);
+            });
+
+            // Assert
+            foreach (var client in Clients)
+            {
+                Assert.True(client.ObjectManager.TryGetObject<BesiegerCamp>(besiegerCampId, out var clientBesiegerCamp));
+                Assert.True(client.ObjectManager.TryGetObject<MobileParty>(besiegerPartyId, out var clientBesiegerParty));
+                Assert.Empty(clientBesiegerCamp._besiegerParties);
+            }
+        }
+
+        [Fact]
+        private void ClientAddBesiegerParty_DoesNothing()
+        {
+            // Arrange
+            var besiegerCamp = ServerCreateObject<BesiegerCamp>(out string besiegerCampId);
+            var besiegerParty = ServerCreateObject<MobileParty>(out string besiegerPartyId);
+            foreach (var client in Clients)
+            {
+                Assert.True(client.ObjectManager.TryGetObject<BesiegerCamp>(besiegerCampId, out var syncedCamp));
+            }
+
+            // Act
+            var firstClient = Clients.First();
+            firstClient.Call(() =>
+            {
+                BesiegerCampCollectionPatches.ListAddOverride(besiegerCamp._besiegerParties, besiegerParty, besiegerCamp);
+            });
+
+            // Assert
+            foreach (var client in Clients.Where(c => c != firstClient))
```

Comment:
Also check on the server
```suggestion
            foreach (var client in Clients.Where(c => c != firstClient).Append(Server))
```

Reply:
Done